File: CS/WindowsApplication3/Main.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.Data.Filtering;


namespace DXSample {
    public partial class Main: XtraForm {
        public Main() {
            InitializeComponent();
        }

        private void OnFormLoad(object sender, EventArgs e) {
           gridControl1.DataSource = GetData();
        }

        private DataTable GetData()
        {
            DataTable dt = new DataTable();
            dt.Columns.Add("ID", typeof(int));
            dt.Columns.Add("Name");
            dt.Columns.Add("Date", typeof(DateTime));

            for (int i = 0; i < 30; i++)
                dt.Rows.Add(i, string.Format("Name {0}", i), DateTime.Now.AddDays(i));

            return dt;
        }

        private void OnShowFilterPopupDate(object sender, DevExpress.XtraGrid.Views.Grid.FilterPopupDateEventArgs e)
        {
            FunctionOperator op = new FunctionOperator("IsDayOff", new OperandProperty(e.Column.FieldName));
            e.List.Add(new DevExpress.XtraEditors.FilterDateElement("IsDayOff", "", op));
        }
    }
}

File: VB/WindowsApplication3/Main.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.Data.Filtering


Namespace DXSample
	Partial Public Class Main
		Inherits XtraForm
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub OnFormLoad(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
		   gridControl1.DataSource = GetData()
		End Sub

		Private Function GetData() As DataTable
			Dim dt As New DataTable()
			dt.Columns.Add("ID", GetType(Integer))
			dt.Columns.Add("Name")
			dt.Columns.Add("Date", GetType(DateTime))

			For i As Integer = 0 To 29
				dt.Rows.Add(i, String.Format("Name {0}", i), DateTime.Now.AddDays(i))
			Next i

			Return dt
		End Function

		Private Sub OnShowFilterPopupDate(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Grid.FilterPopupDateEventArgs) Handles gridView1.ShowFilterPopupDate
			Dim op As New FunctionOperator("IsDayOff", New OperandProperty(e.Column.FieldName))
			e.List.Add(New DevExpress.XtraEditors.FilterDateElement("IsDayOff", "", op))
		End Sub
	End Class
End Namespace
